src/components/bottom-sheet/use-bottom-sheet.tsx

```diff
@@ -0,0 +1,217 @@
+        ...styles.bottomSheet,
+        transform: [
+          {
+            translateY: animatedOffset.interpolate({
+              inputRange: [0, 1],
+              outputRange: [height || -2000, 0],
+            }),
+          },
+        ],
+      }}
+      onLayout={onLayout}
+    >
+      {children}
+    </Animated.View>
+  );
+};
+
+type Props = {
+  isOpen: boolean;
+  close: () => void;
+  animatedOffset: Animated.Value;
+  children: ReactNode;
+};
+
+const InternalBottomSheet = ({
+  isOpen,
+  close,
+  animatedOffset,
+  children,
+}: Props) => {
+  const [height, setHeight] = useState<number>(0);
+  const onLayout = ({nativeEvent}: LayoutChangeEvent) => {
+    setHeight(nativeEvent.layout.height);
+  };
+
+  return (
+    <>
+      <Backdrop isOpen={isOpen} />
+      <ClickableBackground isOpen={isOpen} close={close} height={height} />
+      <AnimatedBottomSheet
+        animatedOffset={animatedOffset}
+        height={height}
+        onLayout={onLayout}
+      >
+        {children}
+      </AnimatedBottomSheet>
+    </>
+  );
+};
+
+export const useBottomSheet = (
+  contentFunction: (close: () => void, focusRef: RefObject<any>) => ReactNode,
+) => {
+  const {theme} = useTheme();
+  const {bottom: safeAreaBottom} = useSafeAreaInsets();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const animatedOffset = useRef(new Animated.Value(0)).current;
+  const focusRef = useRef(null);
+
+  const openSheet = useCallback(
+    () =>
+      Animated.timing(animatedOffset, {
+        toValue: 1,
+        duration: 200,
+        useNativeDriver: true,
+      }).start(),
+    [],
+  );
+
+  const closeSheet = useCallback(
+    () =>
+      Animated.timing(animatedOffset, {
+        toValue: 0,
+        duration: 100,
+        useNativeDriver: true,
+      }).start(),
+    [],
+  );
+
+  const close = () => {
+    closeSheet();
+    setIsOpen(false);
+  };
+  const open = () => {
+    openSheet();
+    setIsOpen(true);
+    const reactTag = findNodeHandle(focusRef.current);
+    if (reactTag) {
+      AccessibilityInfo.setAccessibilityFocus(reactTag);
+    }
+  };
+
+  return {
+    open,
+    isOpen,
+    bottomSheet: (
+      <InternalBottomSheet
+        isOpen={isOpen}
+        close={close}
+        animatedOffset={animatedOffset}
+      >
+        <View
+          style={{
+            marginBottom: Math.max(safeAreaBottom, theme.spacings.medium),
+          }}
+        >
+          {contentFunction(close, focusRef)}
+        </View>
+      </InternalBottomSheet>
+    ),
+  };
+};
+
+const useStyles = StyleSheet.createThemeHook((theme) => ({
+  backdrop: {
+    backgroundColor: 'black',
+    position: 'absolute',
+    bottom: 0,
+    top: 0,
+    left: 0,
+    right: 0,
+  },
+  clickableBackground: {
+    backgroundColor: 'none',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+  },
+  bottomSheet: {
+    backgroundColor: theme.background.level1,
+    width: '100%',
+    position: 'absolute',
```

Comment:
Should this also specify `elevation` for Android?

Reply:
Tried elevation, and saw no difference on Android device, even by comparing screenshots side by side. When searching on the Interwebs I found multiple reports of it not having an effect. I think I'll not invest any time into this as of now.